indexing-service/src/main/java/org/apache/druid/indexing/common/task/Task.java

```diff
@@ -331,4 +331,9 @@ static TaskInfo<TaskIdentifier, TaskStatus> toTaskIdentifierInfo(TaskInfo<Task,
         taskInfo.getTask().getMetadata()
     );
   }
+
+  default boolean doesntNeedLookups()
```

Comment:
The method should rather be called `needsLookups()` or  `loadLookups()`. There is no point naming this method in the negative.

Also, since we want to extend this feature in the future to be able to load lookups on demand, I think we should return an enum here with possible values `NONE`, `ALL`, `ON_DEMAND`.

For the time-being, the behaviour for `ALL` and `ON_DEMAND` would be the same. But later, `ON_DEMAND` can be optimized to load only required lookups.

So, overall:

```java
default LookupLoadingMode needsLookups()
{
   return LookupLoadingMode.ALL;
}
```

Reply:
thanks @kfaraz for the comments. 
Makes sense to me.
I'll ping you when your suggestions will be addressed. 
Also here is something that I've discovered lately and if you have some suggestions/ideas please share them: I've tried to start nano cluster with my changes and discovered that it's not that simple. TaskToolbox injects LookupNodeService, i.e. currently it's not possible to remove it without additional refactoring, so I'm trying to convert it to "Lazy" one by injecting Providing<LookupNodeService> instead so that LookupSerdeModule could create this Provider as well

Reply:
renamed interface @kfaraz